src/Codegen/Constraints/ObjectBuilder.php

```diff
@@ -516,7 +516,7 @@ private function codegenType(
     ?dict<string, SchemaBuilder> $pattern_property_classes = null,
   ): CodegenType {
     if ($property_classes is nonnull) {
-      $required = $this->typed_schema['required'] ?? [];
+      $required = $this->typed_schema['required'] ?? vec[];
```

Comment:
The key required itself is a vec, so the old code had a `(array(empty) | vec<string>)` type. I am guessing it is safe to always have a `vec<string>`.